ChatSecure/Classes/View Controllers/OTRRoomOccupantsViewController.swift

```diff
@@ -10,89 +10,221 @@ import Foundation
 import UIKit
 import PureLayout
 
-public class OTRRoomOccupantsViewController: UIViewController {
+open class OTRRoomOccupantsViewController: UIViewController {
     
-    let tableView: UITableView
-    let viewHandler: OTRYapViewHandler
-    let roomKey: String
+    @IBOutlet open weak var tableView:UITableView!
+    @IBOutlet weak var largeAvatarView:UIImageView!
+    
+    open var viewHandler:OTRYapViewHandler?
+    open var roomKey:String?
+    open var room:OTRXMPPRoom?
+    open var headerRows:[String] = []
+    open var footerRows:[String] = []
     static let CellIdentifier = "Cell"
     
     public init(databaseConnection:YapDatabaseConnection, roomKey:String) {
-        self.tableView = UITableView(frame: .zero, style: .plain)
-        self.roomKey = roomKey
-        viewHandler = OTRYapViewHandler(databaseConnection: databaseConnection)
         super.init(nibName: nil, bundle: nil)
-        setupViewHandler()
+        setupViewHandler(databaseConnection: databaseConnection, roomKey: roomKey)
     }
     
     required public init?(coder aDecoder: NSCoder) {
-        fatalError("Not implemented")
+        super.init(coder: aDecoder)
     }
 
-    private func setupViewHandler() {
-        viewHandler.delegate = self
-        viewHandler.setup(DatabaseExtensionName.groupOccupantsViewName.name(), groups: [roomKey])
+    public func setupViewHandler(databaseConnection:YapDatabaseConnection, roomKey:String) {
+        self.roomKey = roomKey
+        databaseConnection.read({ (transaction) in
+            if let key = self.roomKey {
+                self.room = OTRXMPPRoom.fetchObject(withUniqueID: key, transaction: transaction)
+            }
+        })
+        viewHandler = OTRYapViewHandler(databaseConnection: databaseConnection)
+        if let viewHandler = self.viewHandler, let roomKey = self.roomKey {
+            viewHandler.delegate = self
+            viewHandler.setup(DatabaseExtensionName.groupOccupantsViewName.name(), groups: [roomKey])
+        }
     }
     
-    private func setupTableView() {
-        self.view.addSubview(self.tableView)
-        self.tableView.autoPinEdgesToSuperviewEdges()
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.headerRows.append("name")
+        self.headerRows.append("share")
+        self.headerRows.append("addFriends")
+        self.headerRows.append("mute")
+        self.headerRows.append("members")
+        self.footerRows.append("leave")
+        
+        let image = OTRGroupAvatarGenerator.avatarImage(withUniqueIdentifier: self.roomKey!, width: Int(largeAvatarView.frame.width), height: Int(largeAvatarView.frame.height))
+        largeAvatarView.image = image
+
+        //Setup Table View
+        if (self.tableView == nil) {
```

Comment:
In the earlier version of the code I experienced an unexpected nil crash related to self.tableView being nil even after the init/assignment, which I thought was kind of strange. If this view isn't used/tested outside of the context of a storyboard, I think think the manual path can be left out.